### PR TITLE
[go_router]Show ShellRoute's routes in debug log

### PR DESCRIPTION
**Pull request by koji-1009**
Fixes https://github.com/flutter/flutter/issues/113819

Fix ShellRoute's sub routes to log output when `debugLogDiagnostics` is `true`.
In `example/lib/shell_route.dart`, the following log is output.

```
[GoRouter] Full paths for routes:
             => /a
             =>   /a/details
             => /b
             =>   /b/details
             => /c
             =>   /c/details
```

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/go_router/lib/src/configuration.dart [hunk 1]
```diff
@@ -172,7 +172,9 @@ class RouteConfiguration {
     return 'RouterConfiguration: $routes';
   }
 
-  String _debugKnownRoutes() {
+  /// Returns full paths of all routes and known full paths of route names.
```
Comment (chunhtai) [inline packages/go_router/lib/src/configuration.dart:175]: what's `known full paths of route names`. Can you make it more clear?

File: packages/go_router/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 6.5.6
+
+- Fixed an issue where ShellRoute routes were not logged when debugLogDiagnostic was enabled.
```
[the remark this answers is not in the document]
Comment (koji-1009) [reply packages/go_router/CHANGELOG.md]: Thanks!

File: packages/go_router/lib/src/configuration.dart [hunk 3]
```diff
@@ -172,7 +172,9 @@ class RouteConfiguration {
     return 'RouterConfiguration: $routes';
   }
 
-  /// Returns full paths of all routes and known full paths of route names.
+  /// Returns the full path of [routes].
+  /// A path is prefixed with spaces, depending on the depth of the hierarchy.
+  /// If there are any named routes, they are also appended.
```
Comment (chunhtai) [inline packages/go_router/lib/src/configuration.dart:177]:
```suggestion
  /// Each path is indented based depth of the hierarchy, and its `name`
  /// is also appended if not null
```

**Review by chunhtai — APPROVED** [review]
LGTM

File: packages/go_router/lib/src/configuration.dart (hunk 3 above)
Comment (koji-1009) [reply packages/go_router/lib/src/configuration.dart]: Thanks!

File: packages/go_router/lib/src/configuration.dart [hunk 4]
```diff
@@ -172,7 +172,11 @@ class RouteConfiguration {
     return 'RouterConfiguration: $routes';
   }
 
-  String _debugKnownRoutes() {
+  /// Returns the full path of [routes].
+  /// Each path is indented based depth of the hierarchy, and its `name`
```
Comment (chunhtai) [inline packages/go_router/lib/src/configuration.dart:176]:
```suggestion
  /// Returns the full path of [routes].
  ///
  /// Each path is indented based depth of the hierarchy, and its `name`
```

**Review by chunhtai — COMMENTED** [review]
sorry i missed this one, this should be the last one

**Pull request merged**